lib/RoadizRozierBundle/src/EntityThumbnail/Provider/DocumentThumbnailProvider.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+namespace RZ\Roadiz\RozierBundle\EntityThumbnail\Provider;
+
+use Doctrine\Persistence\ManagerRegistry;
+use RZ\Roadiz\CoreBundle\Entity\Document;
+use RZ\Roadiz\Documents\UrlGenerators\DocumentUrlGeneratorInterface;
+use RZ\Roadiz\RozierBundle\EntityThumbnail\AbstractEntityThumbnailProvider;
+
+/**
+ * Provides thumbnail for Document entities.
+ */
+final class DocumentThumbnailProvider extends AbstractEntityThumbnailProvider
+{
+    public function __construct(
+        private readonly DocumentUrlGeneratorInterface $documentUrlGenerator,
+        private readonly ManagerRegistry $managerRegistry,
+    ) {
+    }
+
+    public function supports(string $entityClass, int|string $identifier): bool
+    {
+        return $this->isClassSupported($entityClass, Document::class);
+    }
+
+    public function getThumbnail(string $entityClass, int|string $identifier): ?array
```

Comment:
```suggestion
    public function getThumbnail(string $entityClass, int|string $identifier): ?EntityThumbnail
```

`getThumbnail` should return a `EntityThumbnail` value object